src/ui/components/Library/Team/View/TestRuns/suspense/TestsCache.ts

```diff
@@ -0,0 +1,79 @@
+import orderBy from "lodash/orderBy";
+import { createCache } from "suspense";
+
+import { GraphQLClientInterface } from "shared/graphql/GraphQLClient";
+import { Test, TestExecution } from "shared/test-suites/TestRun";
+
+import { getTestsGraphQL } from "../../Tests/graphql/TestsGraphQL";
+
+export const testsCache = createCache<
+  [graphQLClient: GraphQLClientInterface, accessToken: string | null, workspaceId: string],
+  Test[]
+>({
+  config: { immutable: true },
+  debugLabel: "testsCache",
+  getKey: ([_, __, workspaceId]) => workspaceId,
+  load: async ([graphQLClient, accessToken, workspaceId]) => {
+    const rawTests = await getTestsGraphQL(graphQLClient, accessToken, workspaceId);
+
+    // Compute the failure rate and add it to the stored test in cache
+    // until we're able to do it from the backend
```

Comment:
Comments like this are easy to overlook and forget about. Add a TODO that links to a Linear issue for this?

Reply:
Sounds good — captured in https://linear.app/replay/issue/SCS-1575/compute-the-failure-rates-in-the-backend-instead-of-the-frontend